lmfit/minimizer.py

```diff
@@ -875,9 +917,14 @@ def scalar_minimize(self, method='Nelder-Mead', params=None, **kws):
         variables = result.init_vals
         params = result.params
 
-        fmin_kws = dict(method=method,
-                        options={'maxiter': 1000 * (len(variables) + 1)})
+        self.set_max_nfev(max_nfev, 1000*(result.nvarys+1))
+        fmin_kws = dict(method=method, options={'maxiter': 2*self.max_nfev})
         fmin_kws.update(self.kws)
+
+        if 'maxiter' in kws:
+            warnings.warn(MAXEVAL_Warning % ('maxiter', thisfuncname()),
+                          RuntimeWarning)
+            kws.pop('maxiter')
```

Comment:
this ```maxiter``` check with a method through ```scalar_minimize``` does not seem to work when using the Model interface

Reply:
Will investigate... I think that ideally we could pop away `maxiter` for all the solvers.  But I'm not confident that we will find all of these....